src/platform/mock-local-home-platform.ts

```diff
@@ -106,39 +37,18 @@ export class MockLocalHomePlatform implements MockUDPListener {
     return this.localDeviceIds;
   }
 
-  private setupUDP() {
-    this.udpScanConfigs.forEach(udpScanConfig => {
-      this.mockNetwork!.registerUDPListener(
-        this,
-        udpScanConfig.listenPort,
-        udpScanConfig.broadcastAddress
-      );
-    });
-  }
-
-  private sendUDPBroadcast(scanConfig: UDPScanConfig) {
-    const packetBuffer = Buffer.from(scanConfig.discoveryPacket, 'hex');
-    this.mockNetwork!.sendUDPMessage(
-      packetBuffer,
-      scanConfig.broadcastPort,
-      scanConfig.broadcastAddress,
-      scanConfig.listenPort,
-      scanConfig.broadcastAddress
-    );
-  }
-
-  // Establish fulfillment path using app code
-  async onUDPMessage(msg: Buffer, rinfo: RemoteAddressInfo): Promise<void> {
-    if (this.app === undefined) {
-      throw new Error('Cannot trigger IdentifyRequest: App was undefined');
+  /**
+   * Takes a `discoveryBuffer` and passes it to the fulfillment app in an `IdentifyRequest`
+   * @param discoveryBuffer  The buffer to be included in the `IdentifyRequest` scan data
+   * @returns  The next localDeviceId registered to the Local Home Platform
+   */
+  public async triggerIdentify(discoveryBuffer: Buffer): Promise<string> {
+    console.debug('Received discovery payload:', discoveryBuffer);
+
+    // Cannot start processing until all handlers have been set on the `App`
+    if (!this.app.areAllHandlersSet()) {
```

Comment:
couldn't we simply test the identifyHandler directly?

Reply:
We could, but this also imposes that `listen()` was called, and that the Execute handler was also set.  I don't know if we want to allow the Identify handler to be set without an Execute handler, since I'm assuming that would be an invalid state for a local fulfillment app.

Reply:
I guess what I find confusing is that the flag is called 'handlerSet', while the error message is about `listen not being called'.

Maybe we could move that get to the platform accessor? You shouldn't be able to get the platform and exercise test functions if `listen` has not be called.

And keep a handler specific check here?

Reply:
I moved the assertion to the `App` side and added a handler-specific check.